Starting points checkout project (#1545)
carpentries/instructor-training: learners/demo_lessons.md

---
title: Lessons for Teaching Demonstrations
---

The list below shows lessons that have been reviewed for good and bad start points to use for a teaching demonstration. When selecting an episode for a
teaching demonstration, trainees should review this list to ensure it is suitable.

> As of July 2023, trainees are allowed to choose the episode from which their 5-minute teaching demo will begin. However,  still we recommend

Reviewer: ```suggestion
As of 14 August 2023, trainees are allowed to choose the episode from which their 5-minute teaching demo will begin. However,  still we recommend
```

---
title: Lessons for Teaching Demonstrations
---

The list below shows lessons that have been reviewed for good and bad start points to use for a teaching demonstration. When selecting an episode for a
teaching demonstration, trainees should review this list to ensure it is suitable.

As of July 2023, trainees are allowed to choose the episode from which their 5-minute teaching demo will begin. However,  still we recommend
reviewing the full lesson during preparation, as this will help provide context for instruction and is also good preparation for teaching a 
complete workshop later on!

If a selected episode is very short and finishes in less than 5 minutes, it may be necessary to continue on to the next episode in that lesson 
to complete a 5 minute demonstration. 

> It is **not** necessary to teach the full episode, or to cover any specific amount of content within the 5 minute period.

Reviewer 2: +1, this seems to be a common point of confusion

**If your preferred lesson is not on the list or includes only ineligible episodes, please choose a different lesson.** 

> If you believe you have a lesson or episode that should be eligible but is not on the list, please email us at [instructor.training@carpentries.org](mailto:instructor.training@carpentries.org).

Reviewer 2: Should they maybe email the session host?

Reviewer: This is something we've gone back and forth with. I think some Trainers are comfortable making this determination but it can cause stress, especially for new Trainers who aren't sure how they should handle these requests. If we have them email the session host, we should provide corresponding guidelines for session hosts (e.g in the handbook).

::::::::::::::::::::::: callout

## Teaching Episodes with Dependencies {#dependencies}

Many Carpentries lessons follow a narrative: an example that runs from episode to episode,
with the same data, files, variables, etc reused and modified as new skills and concepts are taught
to Learners.
This means that the content of later episodes in a lesson is likely to depend on the steps
that have taken place earlier in the lesson.

> The _Recommended Starting Points_ column of the tables below provides a list

Reviewer 3: ```suggestion
The _Recommended Episodes_ column of the tables below provides a list
```

Reviewer: We discussed this at the last meeting, and it's possible that this should change. The rare problem is that occasionally trainees can get through an episode in less than 5 minutes and hence have to continue. This should probably be communicated specifically, though, and may or may not justify keeping this language as-is.

  - [R para Análisis Científicos Reproducibles](https://swcarpentry.github.io/r-novice-gapminder-es/)
  - [Análisis y visualización de datos usando Python](https://www.datacarpentry.org/python-ecology-lesson-es/)

### Data Carpentry: Ecology {#dc-ecology}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

### Data Carpentry: Genomics {#dc-genomics}

If you are an instructor in training and wish to use lessons from Data Carpentry's Genomics curriculum for your teaching demo, please read [these instructions](https://datacarpentry.org/genomics-workshop/teaching_demos/) to be sure you are prepared. You must follow these steps before your teaching demo, or you will be asked to reschedule.

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

| [Introduction to Cloud Computing](https://datacarpentry.org/cloud-genomics/) |  \* [Logging onto  the Cloud](https://datacarpentry.org/cloud-genomics/02-logging-onto-cloud.html)  <br/> \* [Fine tuning your Cloud Setup](https://datacarpentry.org/cloud-genomics/03-verifying-instance.html) | \* [Why of Cloud Computing](https://datacarpentry.org/cloud-genomics/01-why-cloud-computing.html) <br/> \* [Which Cloud for my Data?](https://datacarpentry.org/cloud-genomics/04-which-cloud.html) |  |
| [Data Analysis and Visualization in R](https://datacarpentry.org/genomics-r-intro/) | none | all episodes | This lesson is not yet stable and cannot be used for teaching demonstrations. |

### Data Carpentry: Social Sciences {#dc-socialsci}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

| [Data Analysis and Visualization with R](https://www.datacarpentry.org/r-socialsci/) | \* [Introduction to R](https://datacarpentry.org/r-socialsci/01-intro-to-r.html)<br/>\* [Starting with Data](https://datacarpentry.org/r-socialsci/02-starting-with-data.html)<br/>\* [Data Wrangling with dplyr](https://datacarpentry.org/r-socialsci/03-dplyr.html)<br/>\* [Data Wrangling with tidyr](https://datacarpentry.org/r-socialsci/04-tidyr.html)<br/>\* [Getting Started with R Markdown](https://datacarpentry.org/r-socialsci/06-rmarkdown.html)<br/>\* [Processing JSON Data](https://datacarpentry.org/r-socialsci/07-json.html)<br/> | \* [Before We Start](https://www.datacarpentry.org/r-socialsci/00-intro/) | [Data Visualization with ggplot2](https://datacarpentry.org/r-socialsci/04-ggplot2/) depends on action taken earlier in the lesson - consult [_Teaching Episodes with Dependencies_](#dependencies). |

### Data Carpentry: Geospatial {#dc-geospatial}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

### Data Carpentry: Astronomy {#dc-astro}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

|---------- |--------------- |--------------- |------------------------- |
| [Foundations of Astronomical Data Science](https://datacarpentry.org/astronomy-python/) | \* [Basic Queries](https://datacarpentry.org/astronomy-python/01-query.html) <br/> \* [Coordinate Transformations](https://datacarpentry.org/astronomy-python/02-coords.html) | none |  The other episodes depend on actions taken earlier in the lesson - consult [_Teaching Episodes with Dependencies_](#dependencies). <br/> If using [Basic Queries](https://datacarpentry.org/astronomy-python/01-query.html), skip the episode introduction, *Query Language*, and *Using Jupyter* sections, and begin with [*Connecting to Gaia*](https://datacarpentry.org/astronomy-python/01-query.html#connecting-to-gaia) (`from astroquery.gaia import Gaia`)  |

### Data Carpentry: Image Processing {#dc-image}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

### Library Carpentry {#lc}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

| [Library Carpentry: SQL](https://librarycarpentry.org/lc-sql/) | \* [Selecting and Sorting Data](https://librarycarpentry.org/lc-sql/02-selecting-sorting-data.html)<br/> \* [Filtering](https://librarycarpentry.org/lc-sql/03-filtering.html)<br/> \* [Ordering and Commenting](https://librarycarpentry.org/lc-sql/04-ordering-commenting.html)<br/> \* [Aggregating and Calculating Values](https://librarycarpentry.org/lc-sql/05-aggregating-calculating.html)<br/> \* [Joins and aliases](https://librarycarpentry.org/lc-sql/06-joins-aliases.html)<br/> \* [Saving queries](https://librarycarpentry.org/lc-sql/07-saving-queries.html)<br/> \* [Creating tables and modifying data](https://librarycarpentry.org/lc-sql/09-create.html) | \* [Database design](https://librarycarpentry.org/lc-sql/08-database-design.html)<br/> \* [Other database tools](https://librarycarpentry.org/lc-sql/10-other-tools.html)<br/> \* [Extra-challenges](https://librarycarpentry.org/lc-sql/11-extra-challenges.html)<br/> \* [Good style](https://librarycarpentry.org/lc-sql/Bonus_GoodStyle.html) |  |

### Software Carpentry {#swc}

> |  Lesson   | Recommended Starting Points | [**Not Eligible for Demos**](#ineligible) | Notes

Reviewer 3: ```suggestion
|  Lesson   | Recommended Episodes | [**Not Eligible for Demos**](#ineligible) | Notes
```

| [Version Control with Mercurial](https://swcarpentry.github.io/hg-novice/) |  \* [Configuring Mercurial](https://swcarpentry.github.io/hg-novice/02-configuration/)<br/> \* [Creating a Repository](https://swcarpentry.github.io/hg-novice/03-create-repo/)<br/> \* [Tracking Changes to Files](https://swcarpentry.github.io/hg-novice/04-tracking/) | \* [Automated Version Control](https://swcarpentry.github.io/hg-novice/01-basics/) | Later episodes depend on actions taken earlier in the lesson - consult [_Teaching Episodes with Dependencies_](#dependencies). |

### Lecciones en español {#espanol}

::::::::::::::::::::::: callout

## Enseñanza de episodios con dependencias {#dependencies-es}

Muchas lecciones en *Carpentries* siguen una narrativa común: Un mismo ejemplo
que va de episodio en episodio, con datos, archivos, variables, etc. que son
reutilizados y modificados a medida que nuevas habilidades y conceptos son
enseñados a los alumnos. Esto significa que el contenido de los episodios
dependerá probablemente de los pasos que se hayan dado en episodios anteriores
de una misma lección.

> La columna _Puntos de Partida Recomendados_ en la tabla siguiente proporciona

Reviewer 3: ```suggestion
La columna _Episodios Recomendados_ en la tabla siguiente proporciona
```

una lista de episodios que no dependen en gran medida de ninguno de los
contenidos anteriores de la lección. Si decide enseñar uno de los episodios que
no figuran en esta lista de recomendaciones, deberá asegurarse de que todo está
preparado como si las partes anteriores de la lección ya se hubieran enseñado
antes de comenzar su demostración didáctica.

:::::::::::::::::::::::::::::::

::::::::::::::::::::::: callout

## Episodios no aptos para la demostración didáctica {#ineligible-es}

Algunas lecciones incluyen episodios que **no pueden ser utilizados para una
demostración didáctica**, normalmente porque contienen poca o ninguna
oportunidad para la enseñanza mediante demostraciones en vivo. Estos episodios
se enumeran en la columna _No elegible para demostraciones_ en la tabla
siguiente.

:::::::::::::::::::::::::::::::

> |  Lección   | Puntos de Partida Recomendados | [**No elegible para demostraciones**](#ineligible-es) | Notas

Reviewer 3: ```suggestion
|  Lección   | Episodios Recomendados | [**No elegible para demostraciones**](#ineligible-es) | Notas
```
